### DISCUSSION TITLE
InterSystems IRIS adapter

### DISCUSSION THREAD

**Discussion by isc-tdyar** [apache/superset#23091] [Ideas]
Greetings Superset mavens!

I am creating this issue to request a connector to InterSystems IRIS, a mature database with xDBC and recently DB-API driver. We have a community-developed SQLAlchemy driver "pip install sqlalchemy-iris", and we have free (beer) versions of InterSystems IRIS available for development and testing, see [https://containers.intersystems.com](https://containers.intersystems.com) for containers you can download!

Thanks for any assistance in getting this tested!

Thomas Dyar
Product Manager, ML/AI, InterSystems
thomas.dyar@intersystems.com

**Comment by sfirke** [comment]
Hello!  If you are interested in creating a database connector for InterSystems IRIS, you might find this helpful: https://preset.io/blog/building-database-connector/  In that guide there's also a link to this demo video: https://www.youtube.com/watch?v=lk_sfZq-hrU

Are you proposing building this connector yourself?  I doubt that any of the core Superset developers would be able to take this on themselves but I expect if you get underway with building it and have questions, you can find advice in the Superset community Slack.  You can join that via the link on the main Superset repo page.

**Comment by isc-tdyar** [comment]
OK, thanks for the info!!
Regards,
Thomas

From: Sam Firke ***@***.***>
Date: Tuesday, February 14, 2023 at 4:22 PM
To: apache/superset ***@***.***>
Cc: Thomas Dyar ***@***.***>, Author ***@***.***>
Subject: Re: [apache/superset] InterSystems IRIS adapter (Issue #23071)

Hello! If you are interested in creating a database connector for InterSystems IRIS, you might find this helpful: https://preset.io/blog/building-database-connector/ In that guide there's also a link to this demo video: https://www.youtube.com/watch?v=lk_sfZq-hrU

Are you proposing building this connector yourself? I doubt that any of the core Superset developers would be able to take this on themselves but I expect if you get underway with building it and have questions, you can find advice in the Superset community Slack. You can join that via the link on the main Superset repo page.

**Comment by rusackas** [comment]
Converting this to an Ideas thread since it's more of a feature request than a bug. Having said that, if you (or anyone reading) need further help to create the connector beyond the docs linked, please reach out on Superset Slack and we'll be happy to assist further.

**Comment by sfirke** [comment]
It looks like this has been completed!  https://community.intersystems.com/post/apache-superset-now-iris

Should that be noted in the [list of driver installation instructions](https://superset.apache.org/docs/databases/installing-database-drivers/) or elsewhere?
